### PR TITLE
Removing danging logical axioms on obsoleted relations and add QC

### LINKED ISSUE #752: Make sure QC cries when obsolete terms are used in logical axioms
obsolete rels cant have logical axioms. Why didn't QC say anything?

_Originally posted by @matentzn in https://github.com/oborel/obo-relations/pull/749#discussion_r1296848112_

### PR DESCRIPTION
**Pull request by matentzn**
Fixes #752

### CODE REVIEW

**Comment by anitacaron** [comment]
Checking for more logical axioms on obsolete terms, I found two with the
`TransitiveObjectProperty`. Should we generalise this PR?

**Comment by matentzn** [comment]
Go ahead. Ideally, we combine this with a generalised version of https://github.com/ontodev/robot/blob/master/robot-core/src/main/resources/report_queries/deprecated_property_reference.rq

which takes into account property characteristics and domain and range axioms

File: src/ontology/Makefile [hunk 1]
```diff
@@ -42,9 +42,9 @@ REPORT_FAIL_ON =            ERROR
 REPORT_LABEL =              -l true
 REPORT_PROFILE_OPTS =       --profile $(ROBOT_PROFILE)
 OBO_FORMAT_OPTIONS =        
-SPARQL_VALIDATION_CHECKS =  owldef-self-reference iri-range label-with-iri multiple-replaced_by term-editor-uri dce 
+SPARQL_VALIDATION_CHECKS =  owldef-self-reference iri-range label-with-iri multiple-replaced_by term-editor-uri dce obsolete obsolete-relation
```
Comment (matentzn) [inline src/ontology/Makefile:45]: Why `obsolete obsolete-relation`

Comment (anitacaron) [reply src/ontology/Makefile:45]: Sorry forgot to update repo. The first obsolete is for classes, but we don't have here.

**Comment by matentzn** [comment]
Good to merge!

**Review by anitacaron — APPROVED** [review]

**Review by cthoyt — APPROVED** [review]
lgtm

**Review by cmungall — APPROVED** [review]

**Pull request merged**